src/batList.mli

```diff
@@ -231,6 +231,15 @@ val reduce : ('a -> 'a -> 'a) -> 'a list -> 'a
 
     @raise Invalid_argument on empty list. *)
 
+val fold_map : ('a -> 'b -> ('a * 'c)) -> 'a -> 'b list -> 'a * 'c list
+(** If [f x] is [(f_fst x, f_snd x)], [fold_map f acc [a0; a1; ...;
+    an]] returns both [f_fst (... (f_fst (f_fst acc a0) a1) ...) an]
+    (like [fold_left]) and [[f_snd a0; ...; f_snd an]] (like
+    [map]). Tail recursive.
```

Comment:
I don't understand the comment on first read. Could we have something that is easier to understand? I think that specification in terms of `fold` should be possible, for example, but it would be nice to give an informal textual explanation before.